src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/resource/ReportCorrectionRequestResource.kt

```diff
@@ -115,7 +116,7 @@ class ReportCorrectionRequestResource : ReportRelatedObjectsResource<CorrectionR
         report.addCorrectionRequest(
           reason = reason,
           descriptionOfChange = descriptionOfChange,
-          correctionRequestedBy = authenticationFacade.getUserOrSystemInContext(),
+          correctionRequestedBy = authenticationHolder.username ?: SYSTEM_USERNAME,
```

Comment:
I think we're trying in future to move to explicitly passing through users rather than putting them in the auth token.  That said I haven't seen this pattern before, but wondering if it would be a good idea for the holder to be initialised with a default username in `application.yaml` e.g. `hmpps.auth.default-username` or similar.  Would then have a separate method that would return a `String` instead of nullable `String?` e.g. `holder.usernameOrDefault`

Reply:
i actually feel that the fallback username is a bit of a dirty hack. i've simply kept the logic the same as much as possible for the purpose of this pr. but in my opinion, for the use cases in this repo, we should _require_ the username if we actually expect it.

Reply:
agreed, especially as you'd never notice if a client no longer passes through the `username` in the token.  Hence my comment that it should be passed through explicitly instead.